share/gpodder/extensions/subscription_stats.py

```diff
@@ -0,0 +1,185 @@
+                             Gtk.Label('%d %s' % (len(channels), category)))
+
+    def open_dialog(self):
+        db = self.gpodder.db
+
+        # get all channels
+        channels = []
+        with db.lock:
+            cur = db.cursor()
+            cur.execute(
+                        'SELECT id, title, pause_subscription FROM %s'
+                        % db.TABLE_PODCAST)
+            while True:
+                row = cur.fetchone()
+                if row is None:
+                    break
+                channels.append(row)
+            cur.close()
+
+        # get average time between episodes per channel
+        now = int(time.time())
+        nr_paused = 0
+        daily = []
+        weekly = []
+        monthly = []
+        yearly = []
+        for channel_id, channel_name, paused in channels:
+            if paused:
+                nr_paused += 1
+
+            total = 0
+            nr_episodes = 0
+            prev = now
+            with db.lock:
+                cur = db.cursor()
+                cur.execute('SELECT published FROM %s WHERE podcast_id = %d \
+                            ORDER BY published DESC LIMIT 25'
+                            % (db.TABLE_EPISODE, channel_id))
+                while True:
+                    row = cur.fetchone()
+                    if row is None:
+                        break
+                    if total == 0:
+                        edate = row[0]
+                    total += (prev - row[0])
+                    nr_episodes += 1
+                    prev = row[0]
+                cur.close()
+
+            average = (total / nr_episodes) / (24 * 60 * 60) \
+                if nr_episodes > 0 else 0
+
+            if average == 0:
+                yearly.append([average, channel_name, edate, paused])
+            elif average <= 2:
+                daily.append([average, channel_name, edate, paused])
+            elif average <= 14:
+                weekly.append([average, channel_name, edate, paused])
+            elif average <= 61:
+                monthly.append([average, channel_name, edate, paused])
+            else:
+                yearly.append([average, channel_name, edate, paused])
+
+        # sort by averages
+        daily.sort(key=lambda e: e[0])
+        weekly.sort(key=lambda e: e[0])
+        monthly.sort(key=lambda e: e[0])
+        yearly.sort(key=lambda e: e[0])
+
+        # open dialog
+        dlg = Gtk.Dialog(_('Subscription Statistics'),
+                         self.gpodder.main_window)
+        dlg.set_size_request(400, 400)
+        dlg.set_resizable(True)
+
+        box = Gtk.Box(orientation=Gtk.Orientation.VERTICAL)
+        box.set_border_width(0)
+
+        label = Gtk.Label(_('%d subscriptions (%d paused)') % (len(channels),
+                          nr_paused))
+        label.set_padding(0, 10)
+        box.add(label)
+
+        notebook = Gtk.Notebook()
+        notebook.set_vexpand(True)
+        notebook.set_scrollable(True)
```

Comment:
Adding `notebook.set_show_border(False)` here removes the small but useless border.

Reply:
Done.